addons/web_editor/static/src/xml/editor.xml

```diff
@@ -104,23 +115,80 @@
-                            <input type="text" class="form-control url" id="imageurl" name="url" placeholder="https://www.odoo.com/logo.png" style="width: 250px;"/>
-                            <div class="input-group-append">
-                                <button type="button" class="btn btn-secondary o_upload_image_url_button" disabled="disabled">Add</button>
+                            <input type="text" class="form-control url" id="imageurl" name="url" placeholder="https://www.odoo.com/logo.png" style="width:15rem;"/>
+                            <div class="input-group-append ml-2">
+                                <button type="button" class="btn btn-sm btn-secondary o_upload_image_url_button" disabled="disabled">Add an image URL</button>
                             </div>
                         </div>
+                        <div class="ml-2" style="font-size:21px;">
+                            <span class="warning-url-image text-warning d-none fa fa-warning" title="This URL provided does not contain any image. It will be added in document section."/>
+                            <span class="success-url-image text-success d-none fa fa-check" title="This URL seems valid"/>
+                        </div>
                     </div>
                 </div>
+                <div class="col-sm text-right o_search_image">
+                    <ul class="list-unstyled row">
+                        <li class="nav-item ml-auto search o_search_image_dir">
+                            <div class="form-group">
+                                <input class="form-control o_width_search_bar" id="icon-search" type="search" placeholder="Search an image"/>
+                                <span class="fa fa-search" title="Search" role="img" aria-label="Search"/>
+                                <div class="dropdown-menu o_dropdown_search container-fluid row">
+                                    <a class="dropdown-item o_search_an_image col" href="#" style="font-size:12px;">
+                                        Search among my images for:
+                                        <div class="font-weight-bold o_search_value_input"></div>
+                                    </a>
+                                    <a class="dropdown-item o_search_from_unsplash" href="#" style="font-size:12px;">
+                                        Search from Unsplash for:
+                                        <div class="font-weight-bold o_search_value_input"></div>
+                                    </a>
+                                </div>
+                            </div>
+                        </li>
+                    </ul>
+                </div>
             </div>
-            <div class="form-text"/>
-            <div class="existing-attachments">
-                <t t-call="web_editor.dialog.image.existing"/>
+        </div>
+    </t>
+    <t t-name="web_editor.dialog.document.submenu">
```

Comment:
Seems like that this is the same template as above, or nearly. At best, use generic class names and use the same template. At worst, make this template an extension of the first one (t-extend)